server/routes/spotify.js

```diff
@@ -48,6 +54,31 @@ router.get('/login', (req, res) => {
   res.redirect(`https://accounts.spotify.com/authorize?${queryParams}`)
 })
 
+router.post('/user-login', async (req, res, next) => {
+  const { username, password } = req.body
+
+  if (!username) {
+    return next(new BadRequestError('Missing username field.'))
+  }
+  if (!password) {
+    return next(new BadRequestError('Missing password field.'))
+  }
+
+  /* decrypt hashed password */
+  /* search db for pw that matches correct user */
```

Comment:
I don't think this comment is correct. You're just validating that the password matches for the given user not searching the entire db for a pw that matches. If that were the case that would be a security vulnerability.

Reply:
Woops, idk why I commented this.